README.md

```diff
@@ -1,39 +1,364 @@
+
+  ```dart
+    /// This interface wraps ReviewService<CustomReviewSettings> so that you don't have to repeat the generic parameter everywhere that you would use the review service.
+    /// In other words, you should use this interface in the app instead of ReviewService<CustomReviewSettings> because it's leaner.
+    class CustomReviewService implements ReviewService<CustomReviewSettings> {
+      late ReviewService<CustomReviewSettings> _reviewService;
+
+      CustomReviewService(ReviewService<CustomReviewSettings> reviewService) {
+        _reviewService = reviewService;
+      }
+
+      @override
+      Future<bool> getAreConditionsSatisfied() async {
+        return await _reviewService.getAreConditionsSatisfied();
+      }
+
+      @override
+      Future<void> tryRequestReview() async {
+        await _reviewService.tryRequestReview();
+      }
+
+      @override
+      Future<void> updateReviewSettings(
+        CustomReviewSettings Function(CustomReviewSettings p1) updateFunction,
+      ) async {
+        await _reviewService.updateReviewSettings(updateFunction);
+      }
+    }
+  ```
+
+After creating your own implementation of the ReviewService you just need to register it like this :
+```dart
+  GetIt.I.registerSingleton(
+    CustomReviewService(
+      ReviewService<CustomReviewSettings>(
+        logger: logger,
+        reviewPrompter: ReviewPrompter(logger: logger),
+        reviewSettingsSource: reviewSettingsSource,
+        reviewConditionsBuilder: reviewConditionsBuilder,
+      ),
+    ),
+  );
+```
+
+And use it like that :
+```dart
+  await GetIt.I.get<CustomReviewService>().trackFavoriteJokesCount();
+```
+
+Now you're all set to track how many jokes are favorited in your app and prompt for a review once your review conditions are meant (3 favorite jokes in this case).
+
+## Testing
+
+This is what you need to know before testing and debugging this service. Please note that this may change and you should always refer to the [Apple](https://developer.apple.com/documentation/storekit/skstorereviewcontroller/3566727-requestreview#4278434) and [Android](https://developer.android.com/guide/playcore/in-app-review/test) documentation for the most up-to-date information.
+
+### Android TODO : check if this is still true with the in_app_review package
+
+- You can't test this service while debugging the application, the prompt won't show up. To test it, you need to use the [internal application sharing](https://play.google.com/console/about/internalappsharing/) or the internal testing feature in Google Play Console. See [this](https://developer.android.com/guide/playcore/in-app-review/test) for more details.
+- You can't use a Google Suite account on Google Play to review an application because the prompt will not show up.
+
+### iOS TODO : check if this is still true with the in_app_review package
```

Comment:
There seems to be some TODOs here and just below.

Reply:
The info seems to be correct, I've removed the TODOs